instant-xml/tests/all.rs

```diff
@@ -104,16 +105,223 @@ fn struct_with_custom_field_from_xml() {
+            "<Nested xmlns=\"WRONG\" xmlns:bar=\"BAZ\"><bar:flag>true</bar:flag></Nested>"
+        )
+        .unwrap_err(),
+        Error::WrongNamespace
+    );
+
+    // Correct child namespace
+    assert_eq!(
+        StructWithCorrectNestedNamespace::from_xml("<StructWithCorrectNestedNamespace xmlns=\"URI\" xmlns:bar=\"BAZ\"><Nested xmlns=\"URI\" xmlns:bar=\"BAZ\"><bar:flag>true</bar:flag></Nested></StructWithCorrectNestedNamespace>").unwrap(),
+        StructWithCorrectNestedNamespace {
+            test: Nested { flag: true }
+        }
+    );
+
+    // Correct child namespace - without child redefinition
+    assert_eq!(
+        StructWithCorrectNestedNamespace::from_xml("<StructWithCorrectNestedNamespace xmlns=\"URI\" xmlns:bar=\"BAZ\"><Nested><bar:flag>true</bar:flag></Nested></StructWithCorrectNestedNamespace>").unwrap(),
+        StructWithCorrectNestedNamespace {
+            test: Nested { flag: true }
+        }
+    );
+
+    // Different child namespace
+    assert_eq!(
+        StructWithWrongNestedNamespace::from_xml("<StructWithWrongNestedNamespace xmlns=\"URI\" xmlns:dar=\"BAZ\"><NestedWrongNamespace xmlns=\"\"><flag>true</flag></NestedWrongNamespace></StructWithWrongNestedNamespace>").unwrap(),
+        StructWithWrongNestedNamespace {
+            test: NestedWrongNamespace {
+                flag: true
+            }
+        }
+    );
+
+    // Wrong child namespace
+    assert_eq!(
+        StructWithWrongNestedNamespace::from_xml("<StructWithWrongNestedNamespace xmlns=\"URI\" xmlns:dar=\"BAZ\"><NestedWrongNamespace><flag>true</flag></NestedWrongNamespace></StructWithWrongNestedNamespace>").unwrap_err(),
+        Error::WrongNamespace
+    );
+}
+
+#[derive(Debug, Eq, PartialEq, FromXml)]
+#[xml(namespace("URI", bar = "BAZ"))]
+struct NestedOtherNamespace {
+    #[xml(namespace(bar))]
+    flag: bool,
+}
+
+#[derive(Debug, Eq, PartialEq, FromXml)]
+#[xml(namespace("URI", bar = "BAZ"))]
+struct StructOtherNamespace {
+    test: NestedOtherNamespace,
+}
+
+#[test]
+fn other_namespaces() {
+    // Other namespace not-nested
+    assert_eq!(
+        NestedOtherNamespace::from_xml(
+            "<NestedOtherNamespace xmlns=\"URI\" xmlns:bar=\"BAZ\"><bar:flag>true</bar:flag></NestedOtherNamespace>"
+        )
+        .unwrap(),
+        NestedOtherNamespace { flag: true }
+    );
+
+    // Other namespace not-nested - wrong defined namespace
+    assert_eq!(
+        NestedOtherNamespace::from_xml(
+            "<NestedOtherNamespace xmlns=\"URI\" xmlns:bar=\"BAZ\"><wrong:flag>true</wrong:flag></NestedOtherNamespace>"
+        )
+        .unwrap_err(),
+        Error::UnexpectedPrefix
+    );
+
+    // Other namespace not-nested - wrong parser namespace
+    assert_eq!(
+        NestedOtherNamespace::from_xml(
+            "<NestedOtherNamespace xmlns=\"URI\" xmlns:bar=\"WRONG\"><bar:flag>true</bar:flag></NestedOtherNamespace>"
+        )
+        .unwrap_err(),
+        Error::UnexpectedPrefix
```

Comment:
I think we should yield `Error::WrongNamespace` in all these cases, since prefixes are just syntactic sugar for the actual underlying namespaces.